src/festim/hydrogen_transport_problem.py

```diff
@@ -644,23 +652,28 @@ def create_initial_conditions(self):
                 "Initial conditions can only be defined for transient simulations"
             )
 
-        function_space_value = None
-
         for condition in self.initial_conditions:
-            # create value_fenics for condition
-            function_space_value = None
-            if callable(condition.value):
-                # if bc.value is a callable then need to provide a functionspace
-                if not self.multispecies:
-                    function_space_value = condition.species.sub_function_space
-                else:
-                    function_space_value = condition.species.collapsed_function_space
+            if isinstance(
+                condition, festim.initial_condition.InitialConcentrationFromFile
+            ):
+                u_in = condition.read_function(element="P", order=1)
+                condition.expr_fenics = u_in
```

Comment:
might need to be careful here as its passing an fem.function to expr_fenics, which in all other initial conditions cases is an fem.Expression, however it seems this is fine when it comes to the interpolation.

Reply:
Seems to work fine

Reply:
Seems to work